string visitor_id
string destination

